libraries/core_libs/network/src/tarcap/packets_handlers/pbft_block_packet_handler.cpp

```diff
@@ -26,64 +29,63 @@ void PbftBlockPacketHandler::process(const PacketData &packet_data, const std::s
-  if (pbft_synced_period >= pbft_block->getPeriod()) {
-    LOG(log_tr_) << "Drop new PBFT block " << pbft_block->getBlockHash().abridged() << " at period "
-                 << pbft_block->getPeriod() << ", own PBFT chain has synced at period " << pbft_synced_period;
+  // TODO: need to fix that missing proposal PBFT blocks
+  const auto pbft_chain_size = pbft_chain_->getPbftChainSize();
+  if (pbft_chain_size + 1 != proposed_period) {
+    LOG(log_tr_) << "Drop proposed PBFT block " << proposed_block_hash.abridged() << " at period " << proposed_period
+                 << ", own PBFT chain size is " << pbft_chain_size;
     return;
   }
 
   // Synchronization point in case multiple threads are processing the same block at the same time
   if (!pbft_chain_->pushUnverifiedPbftBlock(pbft_block)) {
-    LOG(log_tr_) << "Drop new PBFT block " << pbft_block->getBlockHash().abridged() << " at period "
-                 << pbft_block->getPeriod() << " -> already inserted in unverified queue ";
+    LOG(log_tr_) << "Drop new PBFT block " << proposed_block_hash.abridged() << " at period " << proposed_period
+                 << " -> already inserted in unverified queue ";
     return;
   }
 
-  LOG(log_dg_) << "Receive proposed PBFT Block " << pbft_block->getBlockHash().abridged()
-               << ", Peer PBFT Chain size: " << peer_pbft_chain_size << " from " << peer->getId();
+  LOG(log_dg_) << "Receive proposed PBFT Block " << proposed_block_hash.abridged() << ", for proposed period "
+               << proposed_period << " from " << peer->getId();
 
   onNewPbftBlock(*pbft_block);
 }
 
-void PbftBlockPacketHandler::onNewPbftBlock(PbftBlock const &pbft_block) {
+void PbftBlockPacketHandler::onNewPbftBlock(const PbftBlock &pbft_block) {
   std::vector<std::shared_ptr<TaraxaPeer>> peers_to_send;
-  const auto my_chain_size = pbft_chain_->getPbftChainSize();
   std::string peers_to_log;
 
   for (auto const &peer : peers_state_->getAllPeers()) {
     if (!peer.second->isPbftBlockKnown(pbft_block.getBlockHash()) && !peer.second->syncing_) {
-      if (!peer.second->isDagBlockKnown(pbft_block.getPivotDagBlockHash())) {
-        LOG(log_tr_) << "sending PbftBlock " << pbft_block.getBlockHash() << " with missing dag anchor"
-                     << pbft_block.getPivotDagBlockHash() << " to " << peer.first;
-      }
       peers_to_send.emplace_back(peer.second);
       peers_to_log += peer.second->getId().abridged();
     }
   }
 
-  LOG(log_dg_) << "sendPbftBlock " << pbft_block.getBlockHash() << " to " << peers_to_log;
+  LOG(log_dg_) << "sendPbftBlock " << pbft_block.getBlockHash() << " with reward period cert votes to " << peers_to_log;
   for (auto const &peer : peers_to_send) {
-    sendPbftBlock(peer->getId(), pbft_block, my_chain_size);
+    // Send reward period cert votes, that include more than reward votes in the proposed PBFT block
+    vote_mgr_->sendRewardPeriodCertVotes(pbft_block.getPeriod() - 1);
```

Comment:
you might not even have the previous pbft block with period `pbft_block.getPeriod() - 1` so what would you send then ?

Reply:
When node propose PBFT block period N, the node must have block with N - 1. 
Node proposal period is always PBFT chain size + 1

Reply:
that is true for proposing blocks, but this has nothing to do with proposing blocks. This is about gossiping blocks - what you just wrote is valid only if the block you received is actually from author of the block. As soon as you receive it from someone else, your statement is not valid

Reply:
You're right. Good catch! Thank you
Fixed. If node has not have period pbft_block.getPeriod() - 1, then just gossip pbft block